add pause function and console printing, fixed restart bug

Alex-Lin5 opened the pull request:
Work is as mentioned on the title and commits

Alex-Lin5 committed:
fixed restart bug, added pause function

Alex-Lin5 committed:
added console print to debug, removed IDE setting files

Subh0m0y commented on src/astrobleme/gui/Main.java:
```diff
@@ -41,8 +41,8 @@
 
-    private static final int WIDTH = 500;
-    private static final int HEIGHT = 500;
+    private static final int WIDTH = 200;
+    private static final int HEIGHT = 200;
```
Why was this necessary? To make it fit on screen or for debugging?

It might also be best to have default values and allow the user to override it through environment variables. (Note that I _did_ write this code over 5 years ago...)

Alex-Lin5 replied:
Yes, small game panel helps debug by reducing time of making way for the snake to eat fruits.

Subh0m0y commented on src/astrobleme/gui/Main.java:
```diff
@@ -62,28 +62,56 @@ public void start(Stage primaryStage) throws Exception {
             Snake snake = grid.getSnake();
             if (loop.isKeyPressed()) {
-                return;
+                if(loop.isRunning()){
+                    System.out.println("Pressed ");
+                    return;
+                }
+                else{
+                }
```
Please use <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>L</kbd> in IntelliJ (or android studio) to format the code automatically.

Also, it might be better to use a logger instead of `System.out.println()`. That way, we can set the log level and turn off unnecessary output when not necessary, like for the end user.

Alex-Lin5 replied:
I will try all suggestion promoted in this page.

Subh0m0y commented on src/astrobleme/gui/Main.java:
```diff
@@ -62,28 +62,56 @@ public void start(Stage primaryStage) throws Exception {
                     break;
                 case RIGHT:
                     snake.setRight();
+                    System.out.println("right ");
```
See above comment on logger and formatting.

Subh0m0y commented on src/astrobleme/gui/Painter.java:
```diff
@@ -58,7 +58,7 @@ public static void paint(Grid grid, GraphicsContext gc) {
         // The score
         gc.setFill(Color.BEIGE);
-        gc.fillText("Score : " + 100 * snake.getPoints().size(), 10, 490);
+        gc.fillText("Score : " + 100 * snake.getPoints().size(), 10, 10);
```
Again, this is a relatively large change in position. It might be better to use an environment variable with a default fallback value here.

Subh0m0y commented on src/astrobleme/logic/GameLoop.java:
```diff
@@ -37,30 +37,39 @@
     private boolean running;
-    private boolean paused;
-    private boolean keyIsPressed;
+    private boolean stopped;
```
I named it paused because it can be potentially extended to support a pause button. Stop takes away that possibility IMO.

Subh0m0y commented on src/astrobleme/logic/GameLoop.java:
```diff
@@ -37,30 +37,39 @@
-        while (running && !paused) {
+        while (true) {
             // Time the update and paint calls
+            if (running);
+            else {
+//                System.out.println("Pausing...... ");
+                try {
+                    Thread.sleep((long) (interval));
+                } catch (InterruptedException ignore){}
+                continue;
+            }
```
Can you briefly explain what this does?

Alex-Lin5 replied:
It is for pause purpose, invoke by keypad 'P'. The event controlls the variable value running.

Subh0m0y commented on src/astrobleme/logic/History.java:
```diff
@@ -0,0 +1,4 @@
+package astrobleme.logic;
+
+public class History {
```
What history will you be storing here? I assume this is for score list?

Alex-Lin5 replied:
It is meant for collecting the snake movement in the past. I am not sure when to start working on that.

Subh0m0y reviewed (changes requested):
Good start! Let's keep iterating on it, and then we can merge it.

Subh0m0y commented:
Also, heads up: @hungrybluedev is my main account. This old account is for archival purposes and generally less active.

Alex-Lin5 committed:
fixed a bug of score is initialized to 100, and removed console printing

Alex-Lin5 commented:
Hi, I managed to debug using default Intellij debuger. I am not sure how to debug using a log, and setting up log level. I fixed the score initialization bug, and removed the console printing.

Alex-Lin5 commented:
Also, I keep the left wall as the hard wall for debug. If the snake head goes to the left edge, the game is over.
